unified-codes/apps/web/src/components/containers/explorer/ExplorerTableRows.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { TableCell, TableRow } from '@unified-codes/ui/components';
+import { withStyles } from '@unified-codes/ui/styles';
+import { EEntityField, IEntity } from '@unified-codes/data';
+
+import { DetailActions, IExplorerAction } from '../../../actions';
+import { ExplorerSelectors } from '../../../selectors';
+import { IState } from '../../../types';
+import { ITheme } from '../../../styles';
+import { TableBody } from '@material-ui/core';
+
+const styles = (theme: ITheme) => {
+  const row = {
+    cursor: 'pointer',
+    '&:hover': { backgroundColor: theme.palette.background.toolbar },
+    '& td:last-child': { borderRight: 0 },
+    '& td:first-child': { fontWeight: 700 },
+  };
+
+  return {
+    root: {},
+    rowPrimary: {
+      ...row,
+      background: theme.palette.background.default,
+    },
+    rowSecondary: {
+      ...row,
+      background: theme.palette.background.paper,
+    },
+    cell: {
+      borderRight: `1px solid ${theme.palette.divider}`,
+      borderBottom: 0,
+    },
+  };
+};
+
+export interface ExplorerTableRowsProps {
+  classes?: {
+    root?: string,
+    rowPrimary?: string,
+    rowSecondary?: string,
+    cell?: string,
+  };
+  columns: EEntityField[];
+  entities: IEntity[];
+  onSelect: (entity: IEntity) => void;
+}
+
+export type ExplorerTableRows = React.FunctionComponent<ExplorerTableRowsProps>;
+
+const ExplorerTableRowsComponent: ExplorerTableRows = ({ classes, columns, entities, onSelect }) => {
+  const history = useHistory();
```

Comment:
I think good convention is to only access navigation state in the page components.  Suggest `onSelect` is passed down as a closure which wraps the `history` object? E.g. in parent component:

```
const onSelect = (entity) => {
  history.push(`/detail/${entity.code$}`;
  dispatch(...);
}
```

Reply:
Sounds reasonable 👍  think I had done something similar with Explorer before (now called `ExplorerPage`). Children should probably not be able to take their parents on unexpected side trips to places of their choosing (stretching the analogy a little).